test/test_trixi2vtk.jl

```diff
@@ -10,25 +11,153 @@ function run_trixi(elixir; kwargs...)
+from the
+[`Trixi2Vtk_reference_files` repository](https://github.com/trixi-framework/Trixi2Vtk_reference_files).
+"""
+function compare_cell_info(out_filename, ref_filename; atol=500*eps(), rtol=sqrt(eps()))
+  ref_vtk = VTKFile(ref_filename)
+  vtk = VTKFile(out_filename)
+
+  # check that the number of cells and points match
+  @test vtk.n_cells == ref_vtk.n_cells
+  @test vtk.n_points == ref_vtk.n_points
+
+  # Compare data information
+
+  # extract the data sets
+  out_cell_data = get_cell_data(vtk)
+  ref_cell_data = get_cell_data(ref_vtk)
 
-function test_trixi2vtk(filenames, outdir; hashes=nothing, kwargs...)
-  @test_nowarn trixi2vtk(joinpath(outdir, filenames); output_directory=outdir, kwargs...)
+  # check that the number of variables and their names match
+  @test length(out_cell_data.names) == length(ref_cell_data.names)
+  @test out_cell_data.names == ref_cell_data.names
 
-  if !isnothing(hashes)
-    for (filename, hash_expected) in hashes
-      hash_measured = sha1file(joinpath(outdir, filename))
-      @test_skip hash_expected == hash_measured
-    end
+  # Note!!
+  # Occassionally, for the last equation variable there is an issue
+  # that the size of the data extracted with
+  # [`ReadVTK.jl`](https://github.com/trixi-framework/ReadVTK.jl)
+  # is one byte larger than expected. Somehow this is related to
+  # the zlib format and/or how
+  # [`WriteVTK.jl`](https://github.com/jipolanco/WriteVTK.jl) created
+  # the file in the first place. Even though there is this access error,
+  # the `vtu` file is valid and can be used with ParaView or VisIt without issue.
+  # Therefore, we check all the variable data except for the last one.
+
+  eqn_check_number = length(ref_cell_data.names) == 1 ? length(ref_cell_data.names) : length(ref_cell_data.names)-1
+
+  # check that the actual plot data is (approximately) the same
+  for i in 1:eqn_check_number
+    variable_name = ref_cell_data.names[i]
+    out_data = get_data(out_cell_data[variable_name])
+    ref_data = get_data(ref_cell_data[variable_name])
+    @test isapprox(out_data, ref_data, atol=atol, rtol=rtol)
   end
 end
 
 
+"""
+    compare_point_info(out_filename, ref_filename; atol=500*eps(), rtol=sqrt(eps()))
+
+Test values from the VTK file header and acutal (possibly interpolated) point data. Uses
+`out_filename` created during testing and compares against `ref_filename` that comes
+from the
+[`Trixi2Vtk_reference_files` repository](https://github.com/trixi-framework/Trixi2Vtk_reference_files).
+"""
+function compare_point_info(out_filename, ref_filename; atol=500*eps(), rtol=sqrt(eps()))
+  # Load the data from both files
+  ref_vtk = VTKFile(ref_filename)
+  vtk = VTKFile(out_filename)
+
+  # Compare header information
+  @test vtk.byte_order == ref_vtk.byte_order
+  @test vtk.compressor == ref_vtk.compressor
+  @test vtk.file_type == ref_vtk.file_type
+  @test vtk.version == ref_vtk.version
```

Comment:
Shouldn't we include this also in `compare_cell_info`?

Alternatively: Should we have a function `compare_headers` and move this out of the specific functions?

Reply:
Weird, I just forgot to put these same tests in the `compare_cell_info`. I can fix this.